profiler/src/ProfilerEngine/Datadog.Profiler.Native/ContentionProvider.cpp

```diff
@@ -152,8 +156,8 @@ void ContentionProvider::AddContentionSample(uint64_t timestamp, uint32_t thread
         // We know that we don't have any span ID nor end point details
 
         rawSample.Timestamp = timestamp;
-        rawSample.Stack.reserve(stack.size());
-        rawSample.Stack.insert(rawSample.Stack.end(), stack.begin(), stack.end());
+        auto end_stack = stack.begin() + std::min(stack.size(), static_cast<std::size_t>(rawSample.Stack.Capacity()));
```

Comment:
Do we need to deal with incomplete call stack?
I mean that we have a limit today (1024?) when we collect the callstack: are we sure that our Stack implementation is using the same? Do we have to add a final frame to tell the backend that the stack is incomplete?

Reply:
We never had to do that so, not sure we should do that today.